app/app.go

```diff
@@ -53,99 +56,42 @@ func New(options ...AppOption) *App {
 	return app
 }
 
-func (a *App) InitServer() {
-	a.srv.AppInitializedOnce.Do(func() {
-		a.initEnterprise()
-
-		a.AddConfigListener(func(oldConfig *model.Config, newConfig *model.Config) {
-			if *oldConfig.GuestAccountsSettings.Enable && !*newConfig.GuestAccountsSettings.Enable {
-				if appErr := a.DeactivateGuests(); appErr != nil {
-					mlog.Error("Unable to deactivate guest accounts", mlog.Err(appErr))
-				}
-			}
-		})
-
-		// Disable active guest accounts on first run if guest accounts are disabled
-		if !*a.Config().GuestAccountsSettings.Enable {
-			if appErr := a.DeactivateGuests(); appErr != nil {
+func (s *Server) FinalizeInit(c *Context) {
+	a := New(ServerConnector(s))
+	s.AddConfigListener(func(oldConfig *model.Config, newConfig *model.Config) {
+		if *oldConfig.GuestAccountsSettings.Enable && !*newConfig.GuestAccountsSettings.Enable {
+			if appErr := a.DeactivateGuests(c); appErr != nil {
 				mlog.Error("Unable to deactivate guest accounts", mlog.Err(appErr))
 			}
 		}
+	})
 
-		// Scheduler must be started before cluster.
-		a.initJobs()
-
-		if a.srv.joinCluster && a.srv.Cluster != nil {
-			a.registerAppClusterMessageHandlers()
+	// Disable active guest accounts on first run if guest accounts are disabled
+	if !*s.Config().GuestAccountsSettings.Enable {
+		if appErr := a.DeactivateGuests(c); appErr != nil {
+			mlog.Error("Unable to deactivate guest accounts", mlog.Err(appErr))
```

Comment:
Gentle reminder to check our logging [principles](https://developers.mattermost.com/contribute/server/style-guide/#log-levels) before merging this change.